test/metabase/pulse_test.clj

```diff
@@ -807,6 +807,20 @@
                     :result (s/pred map?)}
                    (pulse/execute-card {:creator_id (mt/user->id :rasta)} card))))))
 
+(deftest execute-dashboard-test
+  (testing "it runs for each card"
+    (mt/with-temp* [Card          [{card-id-1 :id}]
+                    Card          [{card-id-2 :id}]
+                    Dashboard     [{dashboard-id :id} {:name "Birdfeed Usage"}]
+                    DashboardCard [dashcard-1 {:dashboard_id dashboard-id :card_id card-id-1}]
+                    DashboardCard [dashcard-2 {:dashboard_id dashboard-id :card_id card-id-2}]
+                    User [{user-id :id}]]
+      (let [result (pulse/execute-dashboard {:creator_id user-id} dashboard-id)]
+        (is (= (count result) 2))
+        (is (schema= [{:card   (s/pred map?)
+                       :result (s/pred map?)}]
+                     result))))))
```

Comment:
There were zero tests surrounding this branch of code :open_mouth: 

This one is not the strongest, but at least ensures some sanity. Checking the results more deeply is a little involved and perhaps not worth it…maybe there's another way to get better coverage.

Things will certainly be more straightforward once pulses are ripped out (leaving only dashboard subscriptions)